src/main/java/pdi/project/com/example/recipeapi/service/RecipeService.java

```diff
@@ -1,3 +1,86 @@
 package pdi.project.com.example.recipeapi.service;
 
-public interface RecipeService {}
+import static java.util.Objects.isNull;
+
+import java.sql.Timestamp;
+import java.time.LocalTime;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import pdi.project.com.example.recipeapi.domain.Recipe;
+import pdi.project.com.example.recipeapi.dto.RecipeDTO;
+import pdi.project.com.example.recipeapi.exception.CategoryInvalidExpection;
+import pdi.project.com.example.recipeapi.exception.RecipeValidationException;
+import pdi.project.com.example.recipeapi.repository.*;
+
+@Service
+public class RecipeService {
+
+    private final RecipeRepository recipeRepository;
+    private final CategoryRepository categoryRepository;
+    private final SubCategoryRepository subCategoryRepository;
+    private final StepInstructionService stepInstructionService;
+    private final IngredientService ingredientService;
+
+    @Autowired
+    public RecipeService(
+            RecipeRepository recipeRepository,
+            CategoryRepository categoryRepository,
+            SubCategoryRepository subCategoryRepository,
+            StepInstructionService stepInstructionService,
+            IngredientService ingredientService) {
+        this.recipeRepository = recipeRepository;
+        this.categoryRepository = categoryRepository;
+        this.subCategoryRepository = subCategoryRepository;
+        this.stepInstructionService = stepInstructionService;
+        this.ingredientService = ingredientService;
+    }
+
+    public Recipe addRecipe(RecipeDTO recipeDTO) {
+
+
+        validationRecipeDTO(recipeDTO);
+        Recipe recipe = new Recipe();
+
+        if (isNull(recipeDTO.getCategoryId()) || isNull(recipeDTO.getSubCategoryId())) {
+            throw new CategoryInvalidExpection("Category or Subcategory id cannot be null");
+        }
+        categoryRepository.findById(recipeDTO.getCategoryId()).ifPresent(recipe::setCategory);
+        subCategoryRepository.findById(recipeDTO.getSubCategoryId()).ifPresent(recipe::setSubCategory);
```

Comment:
No caso desses dois métodos anteriores, se não for encontrado a categoria e a sub categoria no banco de dados, devemos prosseguir para salvar a receita sem essa informação? Deveríamos ter um erro informando que não podemos criar uma receita sem categoria e sub categoria?